File: src/app/components/portfolio/project.ts
export interface Project {
    id:string;
    title:string;
    dateRange:string;
    dateRangeDescription:string;
    description:string;
    links?:ProjectLink[];
    tags:string[];
}

export interface ProjectLink {
    title:string;
    url:string;
    openInSameWindow?:boolean;
}

File: Tiptup300.DeweeseDotDev.Portfolio/Projects/Project.cs
namespace Tiptup300.DeweeseDotDev.Portfolio.Projects;

public record Project
(
  string ProjectId,
  string Title,
  string Description,
  IDateRange DateRange,
  IReadOnlyList<IFigure> Figures,
  IReadOnlyList<Link> Links,
  IReadOnlyList<string> Tags
)
{
   public ImageFigure? PrimaryFigure => Figures.FirstOrDefault(x => x is ImageFigure && x.IsPrimary) as ImageFigure;
}
